### PR TITLE
Respect uid/gid sent over cli param or config file

### PR DESCRIPTION
**Pull request by vibhansa-msft**
- uid/gid supplied in config file are respected
- uid/gid supplied over cli params are getting overwritten by the current user and group info during config validation phase.

### CODE REVIEW

**Comment by vibhansa-msft** [comment]
UT addition for this component is not possbile so here is my UT done locally:

vikas@vibhansa-ubn-vm:~/go/src/azure-storage-fuse$ ./blobfuse2 mount /usr/blob_mnt -o allow_other && ls -l /usr/blob_mnt && sudo ./blobfuse2 unmount all
total 0
-rw-r--r-- 1 vikas vikas 5444206592 Oct 25 14:51 tempfile_local.test
Successfully unmounted /usr/blob_mnt
1 of 1 mounts were successfully unmounted

vikas@vibhansa-ubn-vm:~/go/src/azure-storage-fuse$ sudo ./blobfuse2 mount /usr/blob_mnt -o allow_other && ls -l /usr/blob_mnt && sudo ./blobfuse2 unmount all
total 0
-rw-r--r-- 1 root root 5444206592 Oct 25 14:51 tempfile_local.test
Successfully unmounted /usr/blob_mnt
1 of 1 mounts were successfully unmounted

vikas@vibhansa-ubn-vm:~/go/src/azure-storage-fuse$ ./blobfuse2 mount /usr/blob_mnt -o allow_other,uid=1002,gid=1002 && ls -l /usr/blob_mnt && sudo ./blobfuse2 unmount all
total 0
-rw-r--r-- 1 faltoo faltoo 5444206592 Oct 25 14:51 tempfile_local.test
Successfully unmounted /usr/blob_mnt
1 of 1 mounts were successfully unmounted

vikas@vibhansa-ubn-vm:~/go/src/azure-storage-fuse$ ./blobfuse2 mount /usr/blob_mnt -o allow_other,uid=0,gid=0 && ls -l /usr/blob_mnt && sudo ./blobfuse2 unmount all
total 0
-rw-r--r-- 1 root root 5444206592 Oct 25 14:51 tempfile_local.test
Successfully unmounted /usr/blob_mnt
1 of 1 mounts were successfully unmounted

vikas@vibhansa-ubn-vm:~/go/src/azure-storage-fuse$ sudo ./blobfuse2 mount /usr/blob_mnt -o allow_other,uid=1002,gid=1002 && ls -l /usr/blob_mnt && sudo ./blobfuse2 unmount all
total 0
-rw-r--r-- 1 faltoo faltoo 5444206592 Oct 25 14:51 tempfile_local.test
Successfully unmounted /usr/blob_mnt
1 of 1 mounts were successfully unmounted

vikas@vibhansa-ubn-vm:~/go/src/azure-storage-fuse$ sudo ./blobfuse2 mount /usr/blob_mnt -o allow_other,uid=1001,gid=1001 && ls -l /usr/blob_mnt && sudo ./blobfuse2 unmount all
total 0
-rw-r--r-- 1 nonsudo nonsudo 5444206592 Oct 25 14:51 tempfile_local.test
Successfully unmounted /usr/blob_mnt
1 of 1 mounts were successfully unmounted

vikas@vibhansa-ubn-vm:~/go/src/azure-storage-fuse$ sudo ./blobfuse2 mount /usr/blob_mnt -o allow_other,uid=0,gid=0 && ls -l /usr/blob_mnt && sudo ./blobfuse2 unmount all
total 0
-rw-r--r-- 1 root root 5444206592 Oct 25 14:51 tempfile_local.test
Successfully unmounted /usr/blob_mnt
1 of 1 mounts were successfully unmounted

vikas@vibhansa-ubn-vm:~/go/src/azure-storage-fuse$

File: component/libfuse/libfuse.go [hunk 1]
```diff
@@ -188,6 +188,8 @@ func (lf *Libfuse) Validate(opt *LibfuseOptions) error {
 	lf.ignoreOpenFlags = opt.IgnoreOpenFlags
 	lf.nonEmptyMount = opt.nonEmptyMount
 	lf.directIO = opt.DirectIO
+	lf.ownerGID = opt.Gid
+	lf.ownerUID = opt.Uid
```
Comment (souravgupta-msft) [inline component/libfuse/libfuse.go:192]: Can we add some tests for this fix?

**Review by souravgupta-msft — APPROVED** [review]

File: component/libfuse/libfuse.go (hunk 1 above)
Comment (vibhansa-msft) [reply component/libfuse/libfuse.go:192]: We do not have libfuse level UT as this requires full mount and all. But I agree we need to think of some way to at least come up with some config level UT.

Comment (souravgupta-msft) [reply component/libfuse/libfuse.go:192]: E2E or some other mount test. We can think about it and have this some other PR.

File: CHANGELOG.md [hunk 2]
```diff
@@ -6,6 +6,8 @@
 - SAS token is redacted in logs for rename api over dfs endpoint
 - Allow user to configure custom AAD endpoint using MSI_ENPOINT environment variable for MSI based authentication
 - Fail mount if block-cache prefetch count exceeds the defined memory limits.
+- uid/gid supplied as CLI parameters will be shown as actual user/group while listing files.
+- Correctd handling of `umask` libfuse option.
```
Comment (souravgupta-msft) [inline CHANGELOG.md:10]:
```suggestion
- Corrected handling of `umask` libfuse option.
```

**Review by souravgupta-msft — APPROVED** [review]

**Pull request merged**